lib/src/arithmetic/floating_point/floating_point_value.dart

```diff
@@ -446,7 +446,7 @@ class FloatingPointValue implements Comparable<FloatingPointValue> {
         (mantissa.width != other.mantissa.width)) {
       throw Exception('FloatingPointValue widths must match for comparison');
     }
-    final signCompare = sign.compareTo(other.sign);
+    final signCompare = other.sign.compareTo(sign);
```

Comment:
@desmonddak , flipped the sign comparison.

Reply:
More testing is needed as the texponent and mantissa comparisons ignore the sign.

Reply:
Fixed the implementation and added tests.